PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java

```diff
@@ -121,6 +126,34 @@
 
     @Override
     public void appFrameCreated(@NotNull List<String> commandLineArgs) {
+        AppLifecycleListener.super.appFrameCreated(commandLineArgs);
+        initAzureToolkit();
+    }
+
+    @Override
+    public void pluginLoaded(@Nonnull IdeaPluginDescriptor pluginDescriptor) {
+        DynamicPluginListener.super.pluginLoaded(pluginDescriptor);
+        initAzureToolkit();
+    }
+
+    @Override
+    public void pluginUnloaded(@Nonnull IdeaPluginDescriptor pluginDescriptor, boolean isUpdate) {
+        DynamicPluginListener.super.beforePluginUnload(pluginDescriptor, isUpdate);
+        ApacheSenderFactory.INSTANCE.create().close();
+        SDKShutdownActivity.INSTANCE.stopAll(); // shutdown application insights
+    }
+
+    @Override
+    public PluginSettings getSettings() {
+        return settings;
+    }
+
+    @Override
+    public String getPluginId() {
+        return PLUGIN_ID;
+    }
+
+    private void initAzureToolkit() {
```

Comment:
Is this method idempotent? i.e. Can this method(and its calling methods) be called multiple times?